pkg/login/social/connectors/google_oauth.go

```diff
@@ -290,3 +295,17 @@ func (s *SocialGoogle) getGroupsPage(ctx context.Context, client *http.Client, u
 
 	return &data, nil
 }
+
+func (s *SocialGoogle) isHDAllowed(hd string) error {
+	if len(s.info.AllowedDomains) == 0 {
+		return nil
+	}
+
+	for _, allowedDomain := range s.info.AllowedDomains {
+		if hd == allowedDomain {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("the HD claim found in the ID token is not whitelisted by the allowed domains")
```

Comment:
```suggestion
	return fmt.Errorf("the hd claim found in the ID token is not present in the allowed domains")
```

Reply:
Agreed! Changed as requested. Thank you for the suggestion.

Reply:
changed as requested